ros_ws/src/teleoperation/src/keyboard_controller.cpp

```diff
@@ -0,0 +1,87 @@
+#include "keyboard_controller.h"
+
+KeyboardController::KeyboardController()
+{
+    this->drive_parameters_publisher =
+        this->node_handle.advertise< drive_msgs::drive_param >(
+            TOPIC_DRIVE_PARAMETERS, 1);
+}
+
+void KeyboardController::keyboardLoop()
+{
+    std::cout << "Listening to keyboard..." << std::endl;
+    std::cout << "========================" << std::endl;
+
+    double velocity = 0;
+    double angle    = 0;
+    while (ros::ok())
+    {
+        int key = this->getKeyboardCharacter();
+
+        if (key == KEYCODE_W)
+        {
+            velocity += 1;
+        }
+
+        if (key == KEYCODE_S)
+        {
+            velocity -= 1;
+        }
+
+        if (key == KEYCODE_A)
+        {
+            angle += 1;
+        }
+
+        if (key == KEYCODE_D)
+        {
+            angle -= 1;
+        }
+
+        this->publishDriveParameters(velocity, angle);
+    }
+}
+
+int KeyboardController::getKeyboardCharacter()
+{
+    static struct termios old_terminal, new_terminal;
+    // back up current terminal settings
+    tcgetattr(STDIN_FILENO, &old_terminal);
+    new_terminal = old_terminal;
+    // disable buffering
+    new_terminal.c_lflag &= ~ICANON;
+    // apply new settings
+    tcsetattr(STDIN_FILENO, TCSANOW, &new_terminal);
+
+    // read character (non-blocking)
+    int character = getchar();
```

Comment:
is the the `<cstdio>` `getchar()`? Because the `<cstdio>` blocks on input (AFAIK).

Reply:
This will be fixed in #93.